mmdet/datasets/openimages.py

```diff
@@ -0,0 +1,845 @@
+                relationship between the parent class and the child class,
+                of shape (class_num, class_num).
+            parents (list): Corresponding parent class.
+            get_all_parents (bool): Whether get all parent names.
+                Default: True
+
+        Returns:
+            ndarray: The matrix of the corresponding relationship between
+            the parent class and the child class, of shape
+            (class_num, class_num).
+        """
+
+        if 'Subcategory' in hierarchy_map:
+            for node in hierarchy_map['Subcategory']:
+                if 'LabelName' in node:
+                    children_name = node['LabelName']
+                    children_index = self.index_dict[children_name]
+                    children = [children_index]
+                else:
+                    continue
+                if len(parents) > 0:
+                    for parent_index in parents:
+                        if get_all_parents:
+                            children.append(parent_index)
+                        class_label_tree[children_index, parent_index] = 1
+
+                class_label_tree = self._convert_hierarchy_tree(
+                    node, class_label_tree, parents=children)
+
+        return class_label_tree
+
+    def add_supercategory_ann(self, annotations):
+        """Add parent classes of the corresponding class of the ground truth
+        bboxes."""
+        for i, ann in enumerate(annotations):
+            assert len(ann['labels']) == len(ann['bboxes']) == \
+                   len(ann['gt_is_group_ofs'])
+            gt_bboxes = []
+            gt_is_group_ofs = []
+            gt_labels = []
+            for j in range(len(ann['labels'])):
+                label = ann['labels'][j]
+                bbox = ann['bboxes'][j]
+                is_group = ann['gt_is_group_ofs'][j]
+                label = np.where(self.class_label_tree[label])[0]
+                if len(label) > 1:
+                    for k in range(len(label)):
+                        gt_bboxes.append(bbox)
+                        gt_is_group_ofs.append(is_group)
+                        gt_labels.append(label[k])
+                else:
+                    gt_bboxes.append(bbox)
+                    gt_is_group_ofs.append(is_group)
+                    gt_labels.append(label[0])
+            annotations[i] = dict(
+                bboxes=np.array(gt_bboxes).astype(np.float32),
+                labels=np.array(gt_labels).astype(np.int64),
+                bboxes_ignore=ann['bboxes_ignore'],
+                gt_is_group_ofs=np.array(gt_is_group_ofs).astype(np.bool))
+
+        return annotations
+
+    def process_results(self, det_results, annotations,
+                        image_level_annotations):
+        """Process results of the corresponding class of the detection bboxes.
+
+        Will choose to do the following two processing according to the
```

Comment:
Use a better docstring format that forms a list . Remember to preview its effects.
```
Note:
It will choose to do the following two processing according to the parameters: 

1. Whether add parent classes of the corresponding class of the detection bboxes. 
2. Whether ignore the classes that unannotated on that image.
```